waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -20,13 +23,23 @@ printJobMsgsUntilExitReceived chan = do
     J.JobOutput {} -> printJobMessage jobMsg >> printJobMsgsUntilExitReceived chan
     J.JobExit {} -> return ()
 
-migrateDev :: Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
-migrateDev projectDir = do
+-- | Migrates in the generated project context and then copies the migrations dir back
+-- up to the Wasp source dir to ensure they remain in sync.
+migrateDevAndSync :: Path' Abs (Dir WaspProjectDir) -> Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
+migrateDevAndSync waspDir projectDir = do
   chan <- newChan
   (_, dbExitCode) <-
     concurrently
       (printJobMsgsUntilExitReceived chan)
       (DbJobs.migrateDev projectDir chan)
   case dbExitCode of
-    ExitSuccess -> return (Right ())
+    ExitSuccess -> copyMigrationsUp waspDir projectDir
```

Comment:
Made this an implicit part of the migration process so we don't have to remember to call it as a caller to this function.